### PR TITLE
Makes it possible to cast a document to a value.

### PR DESCRIPTION
**Pull request by lemire**
This might ease the work of people writing wrappers and so forth.

Note that this was de facto possible with `at_pointer("")`. Except that it was buggy because scalar documents cannot be used as `value` instances. Now we check and throw an error.

credit to @NicolasJiaxin for the idea

### CODE REVIEW

**Comment by lemire** [comment]
@NicolasJiaxin Would you review ?

**Comment by NicolasJiaxin** [comment]
This is really nice! I think this could prove useful elsewhere than with my issue. Thanks for this

**Review by NicolasJiaxin — APPROVED** [review]
Should we also add `get_doc(value)` to revert back from value to doc? Of course there would be checks for depth.

**Comment by lemire** [comment]
@NicolasJiaxin 

> Should we also add get_doc(value) to revert back from value to doc? Of course there would be checks for depth.

Can you elaborate? Maybe provide an example?

By convention, right now, you should only ever have just one document instance alive at any one time. While you are accessing the document, the `document` instance should remain in scope: it is your "iterator" which keeps track of where you are in the document. So if you have a value instance, then you have a document instance somewhere. You could, of course, create a new document instance from a value or an array or an object, but then you'd have two document instances and we do not want to do that... (it would almost surely create nasty bugs).

So it is deliberate that you cannot easily create new documents from one existing document.

If you are ever in a state where you have a `value` (or `array` or `object`) instance and you have lost the document instance (e.g., it has been destroyed) then you effectively have a dangerous bug. So in the same way we keep track of the `parser` instance, we also need to keep track of the current `document` instance. Now, it is possible for the document to be still explicitly in scope, so, for example, I could do...

```
f(iterate(json).get_value())
```

But if the function `f` needs a document instance, then... well, then why not pass/move a document instance instead?

(This is not me dismissing your query, I am just explaining why I don't see the need.)

**Comment by lemire** [comment]
Note that in my exemple...

```
f(iterate(json).get_value())
```

If the function `f` stores that `value` instance somewhere, where it is later accessed, after the document instance has been destroyed, then you are in danger territory. You can do that with On Demand.

**Comment by NicolasJiaxin** [comment]
Yes, never mind my query. I forgot that the document instance is and should be kept alive even after using `get_value`, so you are probably right that there is no need to get a second document from a value instance (plus it is dangerous as you have mentionned).

**Comment by lemire** [comment]
This has been merged.

cc @NicolasJiaxin

File: include/simdjson/generic/ondemand/value-inl.h [hunk 1]
```diff
@@ -144,6 +144,13 @@ simdjson_really_inline simdjson_result<json_type> value::type() noexcept {
   return iter.type();
 }
 
+simdjson_really_inline simdjson_result<bool> value::scalar() noexcept {
```
Comment (jkeiser) [inline include/simdjson/generic/ondemand/value-inl.h:147]: I know I'm way late to the party, but we should be careful to be consistent about boolean naming or things will get hard to use real fast. Should it be `is_scalar` or do we omit `is` in other places?

**Comment by jkeiser** [comment]
Oh! This is a really useful and nice idea :) Scalar documents are a major aberration anyway.

**Pull request merged**